### PR TITLE
Update InstanceID to listen to a notification from Messaging

### PR DESCRIPTION
**Pull request by chliangGoogle**
When we release Messaging that removes InstanceID dependency, Messaging will handle FCM token on its own. However, developers might still depend on both Messaging and InstanceID, each has its own cache for token, but share the same storage(keychain). We need to add a method in InstanceID to listen to a notification from Messaging for token refresh, if that happens, InstanceID will update its cache, but not storage, as Messaging already updated it. 

We want developers to adopt this version before hand as we can't re-enforce iid version once messaging remove it from its dependency. So I updated the new version of messaging to depend on at least this version of IID.

Manual tested in the chen/fm-master branch which has the latest version of Messaging without IID dependency.
Add/delete token from Messaging API and try getToken in InstanceID should return the same token and FID.

Minor: 
Fixed an issue that the local defaultFCMToken is not updated when calling tokenWithAuthorizedEntity:scope:options:handler. This cause developers won't get notified of token update. It is not discovered probably because not so many developers use this API.

### CODE REVIEW

File: Firebase/InstanceID/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 2020-08 -- 4.5.2
```
Comment (paulb777) [inline Firebase/InstanceID/CHANGELOG.md:1]: Should be 4.6.0 with the new functionality

File: FirebaseInstanceID.podspec [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name             = 'FirebaseInstanceID'
-  s.version          = '4.5.1'
+  s.version          = '4.5.2'
```
Comment (paulb777) [inline FirebaseInstanceID.podspec:3]: 4.6.0

File: FirebaseMessaging.podspec [hunk 3]
```diff
@@ -55,7 +55,7 @@ device, and it is completely free.
   s.osx.framework = 'SystemConfiguration'
   s.weak_framework = 'UserNotifications'
   s.dependency 'FirebaseCore', '~> 6.10'
-  s.dependency 'FirebaseInstanceID', '~> 4.3'
+  s.dependency 'FirebaseInstanceID', '~> 4.5', '>= 4.5.2'
```
Comment (paulb777) [inline FirebaseMessaging.podspec:58]: '~> 4.6'

File: Firebase/InstanceID/CHANGELOG.md (hunk 1 above)
Comment (chliangGoogle) [reply Firebase/InstanceID/CHANGELOG.md]: There's no public API change, I guess this is for better version managing?

File: FirebaseMessaging.podspec (hunk 3 above)
Comment (chliangGoogle) [reply FirebaseMessaging.podspec]: Done.

File: Firebase/InstanceID/CHANGELOG.md (hunk 1 above)
Comment (paulb777) [reply Firebase/InstanceID/CHANGELOG.md]: Minor version updates aren't just for public API changes - they should be used for any changes that go beyond bug fixes.  See https://semver.org/

File: Firebase/InstanceID/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+# 2020-08 -- 4.6.0
+- [added] Added a new notification listening token refresh from Messaging and update its cache. (#6286)
```
Comment (paulb777) [inline Firebase/InstanceID/CHANGELOG.md:2]: `its` is ambiguous. Do you mean the IID cache?

Comment (chliangGoogle) [reply Firebase/InstanceID/CHANGELOG.md]: Done. I updated the changelog to clarify it.

**Pull request merged**